lynxkite-graph-analytics/src/lynxkite_graph_analytics/pykeen_ops.py

```diff
@@ -0,0 +1,331 @@
+    bundle = bundle.copy()
+    model = bundle.other.get("model")
+    pred = predict_triples(
+        model=model,
+        triples=TriplesFactory.from_labeled_triples(
+            bundle.dfs[table_name][["head", "relation", "tail"]].values,
+            create_inverse_triples=req_inverse_triples(model),
+        ),
+    )
+    df = pred.process(
+        factory=TriplesFactory.from_labeled_triples(
+            bundle.dfs["triples_test"][["head", "relation", "tail"]].values,
+            create_inverse_triples=req_inverse_triples(model),
+        )
+    ).df
+    bundle.dfs["pred"] = df
+    return bundle
+
+
+@op("Target Prediction")
+def target_predict(bundle: core.Bundle, *, head: str, relation: str, tail: str):
+    """
+    Leave the target prediction field empty
+    """
+    bundle = bundle.copy()
+    model = bundle.other.get("model")
+    pred = predict_target(
+        model=model,
+        head=head if head != "" else None,
+        relation=relation if relation != "" else None,
+        tail=tail if tail != "" else None,
+        triples_factory=TriplesFactory.from_labeled_triples(
+            bundle.dfs["triples_train"][["head", "relation", "tail"]].values,
+            create_inverse_triples=req_inverse_triples(model),
+        ),
+    )
+
+    pred_annotated = pred.add_membership_columns(
+        validation=TriplesFactory.from_labeled_triples(
+            bundle.dfs["triples_val"][["head", "relation", "tail"]].values,
+            create_inverse_triples=req_inverse_triples(model),
+        ),
+        testing=TriplesFactory.from_labeled_triples(
+            bundle.dfs["triples_test"][["head", "relation", "tail"]].values,
+            create_inverse_triples=req_inverse_triples(model),
+        ),
+    )
+    bundle.dfs["pred"] = pred_annotated.df
+
+    return bundle
+
+
+@op("Full prediction", slow=True)
+def full_predict(bundle: core.Bundle, *, k: int = None):
+    """Pass k=0 to keep all scores"""
+    bundle = bundle.copy()
+    model = bundle.other.get("model")
+    pred = predict_all(model=model, k=k if k != 0 else None)
+    pack = pred.process(
+        factory=TriplesFactory.from_labeled_triples(
+            bundle.dfs["triples_val"][["head", "relation", "tail"]].values,
+            create_inverse_triples=req_inverse_triples(model),
+        )
+    )
+    pred_annotated = pack.add_membership_columns(
+        training=TriplesFactory.from_labeled_triples(
+            bundle.dfs["triples_train"][["head", "relation", "tail"]].values,
+            create_inverse_triples=req_inverse_triples(model),
+        )
+    )
+    bundle.dfs["pred"] = pred_annotated.df
+
+    return bundle
+
+
+@op("Extract embeddings from PyKEEN model")
+def extract_from_pykeen(bundle: core.Bundle, *, node_embedding_name: str, edge_embedding_name: str):
+    bundle = bundle.copy()
+    model = bundle.other["model"]
+    triples = TriplesFactory.from_labeled_triples(
+        bundle.dfs["triples_train"][["head", "relation", "tail"]].values,
```

Comment:
Take table name as a parameter.

Must this be the same table as the one used for the training? If it can only be used with that, then perhaps it's best to extract these at the end of training, as part of the training box.